CMakeLists.txt

```diff
@@ -52,11 +52,120 @@ option(FL_BUILD_EXPERIMENTAL "Build internal experimental components" OFF)
 option(FL_BUILD_SCRIPTS "Build internal scripts for wav2letter++" OFF)
 option(FL_BUILD_RECIPES "Build recipes" ON)
 option(FL_BUILD_STANDALONE "Build standalone installation" ON)
-option(FL_BUILD_LIBRARIES "Build flashlight libraries" ON)
+
+
+# --------------------------- Lib Options  ---------------------------
+# Build all libs by default. In order to selectively build libs,
+# disable this option and opt-in to each using the individual build
+# options. For example, if you are only building core, you only need
+# the common library, so you can use -DFL_BUILD_ALL_LIBS=OFF and
+# -DFL_BUILD_LIB_COMMON_ON
+option(FL_BUILD_ALL_LIBS "Build flashlight libraries" ON)
+option(FL_BUILD_LIB_SET "Build flashlight set library" ${FL_BUILD_ALL_LIBS})
+option(FL_BUILD_LIB_SEQUENCE "Build flashlight sequence library" ${FL_BUILD_ALL_LIBS})
+option(FL_BUILD_LIB_AUDIO "Build flashlight audio library" ${FL_BUILD_ALL_LIBS})
+option(FL_BUILD_LIB_COMMON "Build flashlight common library" ${FL_BUILD_ALL_LIBS})
+option(FL_BUILD_LIB_TEXT "Build flashlight text library" ${FL_BUILD_ALL_LIBS})
+
+# --------------------------- Core Options  ---------------------------
 option(FL_BUILD_CORE "Build flashlight core" ON)
-cmake_dependent_option(FL_BUILD_CONTRIB
-  "Build and link additional flashlight contrib assets." ON
-  "FL_BUILD_CORE" OFF)
+
+# Small utility function which wraps cmake_dependent options and throws an error if the user
+# is trying to build something without the proper dependencies.
+macro(FL_DEPENDENT_OPTION OPTION TEXT VAL DEPS FRCE)
```

Comment:
1. Will definitely do. Good catch. 
2. The problem with using function is that the scope is not global, so we can't actually perform the check on the variables that are passed in because they only exist on the global scope. (And I think setting the option will also only be for the local scope, but not sure).